watchdog/src/printer.rs

```diff
@@ -0,0 +1,36 @@
+use std::panic;
+
+/// Prints a message to the console.
+pub fn print(msg: &str) {
+    #[cfg(target_arch = "wasm32")]
+    ic_cdk::api::print(msg);
+
+    #[cfg(not(target_arch = "wasm32"))]
+    println!("{}", msg);
+}
+
+/// Sets a custom panic hook, uses debug.trace
+pub fn set_panic_hook() {
```

Comment:
Personal opinion: having to copy this because you want to use your own print function seems like an overkill. Maybe just use the cdk version but guard the code with `#[cfg(target_arch = "wasm32")]` where you need to install the hook? I guess it mainly makes sense for running in a canister anyway, not in a native unit test.